src/cpp/core/system/PosixShellUtils.cpp

```diff
@@ -23,8 +23,8 @@ namespace shell_utils {
 
 std::string escape(const std::string& arg)
 {
-   boost::regex pattern("(\\\\|\\$|`|!|\n|\")");
-   return "\"" + boost::regex_replace(arg, pattern, "\\\\$1") + "\"";
+   boost::regex pattern("[\\\\$`!\n\"]");
```

Comment:
The changes here just tweak the pattern to use the original `[]`-enclosed characters as per @jcheng5's initial comment.